server/chat-plugins/the-studio.ts

```diff
@@ -0,0 +1,613 @@
+			buf += `<br /><strong>${rec.userData.name}</strong></span></td>`;
+		}
+		buf += `</tbody></table>`;
+		buf += `</div>`;
+		return buf;
+	}
+
+	likeRecommendation(artist: string, title: string, liker: User) {
+		const rec = this.get(artist, title);
+		if (!rec) {
+			throw new Chat.ErrorMessage(`The song titled '${title}' by ${artist} isn't recommended.`);
+		}
+		if (!rec.liked) {
+			rec.liked = {ips: [], userids: []};
+		}
+		if (rec.liked.ips.includes(liker.latestIp) || rec.liked.userids.includes(liker.id)) {
+			throw new Chat.ErrorMessage(`You've already liked this recommendation.`);
+		}
+		rec.likes++;
+		rec.liked.ips.push(liker.latestIp);
+		rec.liked.userids.push(liker.id);
+		saveRecommendations();
+	}
+
+	get(artist: string, title: string, submitter?: string, fromSuggestions = false) {
+		let recs = recommendations.saved;
+		if (fromSuggestions) recs = recommendations.suggested;
+		return recs.find(x => toID(x.artist) === toID(artist) &&
+			toID(x.title) === toID(title) &&
+			(submitter ? toID(x.userData.name) === toID(submitter) : true));
+	}
+
+	getIndex(artist: string, title: string, submitter?: string, fromSuggestions = false) {
+		let recs = recommendations.saved;
+		if (fromSuggestions) recs = recommendations.suggested;
+		const index = recs.findIndex(x => toID(x.artist) === toID(artist) &&
+				toID(x.title) === toID(title) &&
+				(submitter ? toID(x.userData.name) === toID(submitter) : true));
+		return index;
+	}
+}
+
+export const LastFM = new LastFMInterface();
+export const Recs = new RecommendationsInterface();
+
+export const commands: ChatCommands = {
+	registerlastfm(target, room, user) {
+		if (!target) return this.parse(`/help registerlastfm`);
+		this.checkChat(target);
+		target = this.filter(target) || '';
+		if (!target) {
+			throw new Chat.ErrorMessage(`The provided account name has phrases that PS doesn't allow.`);
+		}
+		LastFM.validateAccountName(target);
+		this.sendReply(LastFM.addAccountName(user.id, target.trim()));
+	},
+	registerlastfmhelp: [
+		`/registerlastfm [username] - Adds the provided [username] to the last.fm database for scrobbling.`,
+		`Usernames can only be 2-15 characters long, must start with a letter, and can only contain letters, numbers, hyphens, and underscores.`,
+	],
+
+	async lastfm(target, room, user) {
+		this.runBroadcast(true);
+		this.splitTarget(target, true);
+		const username = LastFM.getAccountName(target ? target : user.name);
+		this.sendReplyBox(
+			await LastFM.getScrobbleData(username, this.targetUsername ? this.targetUsername : user.named ? user.name : undefined)
+		);
+	},
+	lastfmhelp: [
+		`/lastfm [username] - Displays the last scrobbled song for the person using the command or for [username] if provided.`,
+		`To link up your last.fm account, check out "/help registerlastfm".`,
+	],
+
+	async track(target, room, user) {
+		if (!target) return this.parse('/help track');
+		this.checkChat(target);
+		const [track, artist] = this.splitOne(target);
```

Comment:
```suggestion
		const [track, artist] = Utils.splitFirst(target, ',');
```
I'm deprecating ``this.splitOne`` soon.

Reply:
Actually maybe don't. I forgot that `splitOne` has other features like defaulting to `,` and trimming.

Reply:
@mia-pi-git In general, there's no need to comment on PRs to remove things _before_ they're deprecated. The deprecation PR itself explains exactly what approach you want to change to and why; without it, these sorts of change suggestions are unnecessary distractions.